tools/picard.wdl

```diff
@@ -1045,3 +1043,189 @@ task create_sequence_dictionary {
+                        "ATTRIBUTE_TO_REVERSE_COMPLEMENT=",
+                        attributes_to_reverse_complement
+                    )
+                )
+            } \
+            KEEP_FIRST_DUPLICATE=~{keep_first_duplicate} \
+            ~{if defined(library_name) then "LIBRARY_NAME=" + library_name else ""} \
+            MAX_DISCARD_FRACTION=~{max_discard_fraction} \
+            REMOVE_ALIGNMENT_INFORMATION=~{remove_alignment_information} \
+            REMOVE_DUPLICATE_INFORMATION=~{remove_duplicate_information} \
+            RESTORE_HARDCLIPS=~{restore_hardclips} \
+            RESTORE_ORIGINAL_QUALITIES=~{restore_original_qualities} \
+            ~{if defined(sample_alias) then "SAMPLE_ALIAS=" + sample_alias else ""} \
+            SANITIZE=~{sanitize} \
+            SORT_ORDER=~{sort_order} \
+            VALIDATION_STRINGENCY=SILENT
+    >>>
+
+    output {
+        File reverted_bam = "~{prefix}.reverted.bam"
+    }
+
+    runtime{
+        memory: "~{memory_gb} GB"
+        disks: "~{disk_size_gb} GB"
+        container: "quay.io/biocontainers/picard:3.1.1--hdfd78af_0"
+        maxRetries: 1
+    }
+}
+
+task fastq_to_sam {
+    meta {
+        description: "This WDL task generates an unaligned BAM (uBAM) from FASTQ files."
+        outputs: {
+            unaligned_bam: "The unaligned BAM file"
+        }
+    }
+
+    parameter_meta {
+        read_one_fastq_gz: "Gzipped FASTQ file with 1st reads in pair"
+        read_two_fastq_gz: "Gzipped FASTQ file with 2nd reads in pair"
+        prefix: "Prefix for the output file. The extension `.unmapped.bam` will be added."
+        read_group_name: "Name of the read group"
+        sample_name: "Sample name for the fastq files"
+        library_name: "Library name"
+        sequencing_center: "Sequencing center that produced the reads"
+        run_date: "Date the run was produced"
+        platform_unit: "Platform unit (e.g. flowcell-barcode.lane for Illumina)"
+        platform: "Platform/technology used to produce the reads (e.g. ILLUMINA)"
+        platform_model: "Model of the sequencing platform (e.g. HiSeq2000). Free-form text."
+        memory_gb: "RAM to allocate for task, specified in GB"
+        modify_disk_size_gb: "Add to or subtract from dynamic disk space allocation. Default disk size is determined by the size of the inputs. Specified in GB."
+    }
+
+    input {
+        File read_one_fastq_gz
+        File read_two_fastq_gz
+        String read_group_name
+        String sample_name
+        String library_name
+        String sequencing_center
+        String run_date
+        String prefix = basename(read_one_fastq_gz, ".fq.gz")
+        String platform_unit = ""
+        String platform = ""
+        String platform_model = ""
+        Int memory_gb = 8
+        Int modify_disk_size_gb = 0
+    }
+
+    Float fastq_size = size(read_one_fastq_gz, "GiB") + size(read_two_fastq_gz, "GiB")
+    Int disk_size_gb = ceil(fastq_size * 2) + 10 + modify_disk_size_gb
```

Comment:
Is this calculation solid? I'm not really sure what to expect for the size of a uBAM.

Reply:
I'd expect them to be around the same size as the FASTQ files. Since they're name sorted (since they're unaligned), you're not going to gain compression from shared sequence (most likely). I'm not sure if we should have additional buffer.